source/agora/node/API.d

```diff
@@ -198,4 +200,75 @@ public interface API
     ***************************************************************************/
 
     public Hash[] getMerklePath (ulong block_height, Hash hash);
+
+
+    /***************************************************************************
+
+        Get UTXOs from the UTXO set
+
+        API:
+            GET /get_utxos
+
+        Params:
+            address = used to identify the owner of UTXOs
+            type = transaction type of UTXOs to find
+            min_amount_value = mininum amount of UTXOs to find
+
+        Returns:
+            the associative array for UTXOs
+
+    ***************************************************************************/
+
+    public UTXOSetValue[Hash] getUTXOs (PublicKey address, TxType type,
+        ulong min_amount_value);
+
+
+    /***************************************************************************
+
+        Create an enrollment data for enrollment process
+
+        Params:
+            frozen_utxo_hash = the hash of a frozen UTXO used to identify
+                        a validator or an enrollment data
+
+        Returns:
+            the Enrollment object created
+
+    ***************************************************************************/
+
+    public Enrollment createEnrollment (Hash frozen_utxo_hash);
```

Comment:
And if the utxo doesn't exist, what happens then?

Reply:
Your question reminds me that I should check the validity of the enrollment data in the receiveEnrollment function of the GossipProtocol class and the validity of the UTXO in the createEnrollment function that you mentioned above.

Reply:
Herny is making the function for checking validity of UTXO called isInvalidValidatorUTXO. So I will use it after his work is done.